### ISSUE TITLE
[BUG] <isFull set true, host and port don't auto register selector info>

### ISSUE THREAD

**Issue by wangtao960807** [apache/shenyu#4674]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

<img width="870" alt="image" src="https://github.com/apache/shenyu/assets/130641415/658287a7-589a-4e89-b27e-d5a8db895463">

### Expected Behavior

host and port existed

### Steps To Reproduce

```yaml
shenyu:
  client:
    http:
      props:
        appName: ksh-api
        contextPath: ${server.servlet.context-path}
        host: 172.16.41.189
        port: 8080
        isFull: true
  register:
    registerType: http
    serverLists: http://172.16.41.9:9095
    props:
      username: admin
      password: 123456
```

### Environment

```markdown
ShenYu version(s): 2.5.1
```

**Comment by peijxx** [comment]
SpringMvcClientEventListener
![image](https://github.com/apache/shenyu/assets/130011144/f2db408c-7ceb-4cfa-bf35-1bed5493fb1c)


AbstractContextRefreshedEventListener
![image](https://github.com/apache/shenyu/assets/130011144/beeb180b-1333-4b08-8313-bbbb413c3bad)

when isFull = true，Not executed “publisher.publishEvent(buildURIRegisterDTO(context, beans));”

**Comment by moremind** [comment]
this similar to https://github.com/apache/shenyu/issues/4512

**Comment by moremind** [comment]
Hi, it will fix in #4681
